### PR TITLE
[16.0][IMP] account_payment_order: check that bank allows out-payments

### PR DESCRIPTION
**Pull request by astirpe**
Add check that bank allows out-payments

### CODE REVIEW

File: account_payment_order/models/account_move.py [hunk 1]
```diff
@@ -153,6 +154,17 @@ def create_account_payment_line(self):
...
                 )
+
+            # Check that the bank allows out payments
+            for line in applicable_lines.filtered(
+                lambda l: l.account_id.account_type == "liability_payable"
+            ):
+                bank = line.partner_bank_id or first(line.partner_id.bank_ids)
+                if bank and not bank.allow_out_payment:
+                    raise UserError(
+                        _("Bank %s does not allow out payments") % bank.bank_name
```
Comment (alexis-via) [inline account_payment_order/models/account_move.py]: I suggest a better error message:
The option "Send Money" is not enabled on the bank account %(bank_account)s of partner %(partner)s.

**Review by flotho — APPROVED** [review]
code review, thanks LGTM

**Comment by flotho** [comment]
Hi @astirpe cuold you re-trigger runboat with a commit ?

**Review by pedrobaeza — APPROVED** [review]
.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Wait, I'm seeing that this is not enough, as we should filter these accounts in:

- Automatic selection when no bank is selected in the invoice.
- When displaying the account in the invoice if the payment mode has this configuration.

**Comment by sbidoul** [comment]
I only now notice this implements https://github.com/OCA/bank-payment/issues/1330

To me this PR looks good. I think it's fine to let users prepare payment orders, and block when generating.

@astirpe could you rebase ?

**Review by pedrobaeza — APPROVED** [review]
OK, let's go with this option

/ocabot merge patch

**Comment by pedrobaeza** [comment]
Please fw-port it to 17.0

**Comment by pedrobaeza** [comment]
Uhm, it seems there are affected tests by this change. Can you please rebase and check them?

**Comment by pedrobaeza** [comment]
Superseded by #1340

**Pull request closed without merging**